### PR TITLE
feat(profiling): Expand the discard reasons

### PR DESCRIPTION
**Pull request by phacops**
We only have a limited number of discard reasons for profiling and as the number of invalid profiles grow, it's difficult to understand why and make sure it's not due to a bug.

This PR aims to add more discard reasons and log an error in case we have a JSON deserialization error so we can investigate and fix it.

### CODE REVIEW

File: relay-profiling/src/sample.rs [hunk 1]
```diff
@@ -233,6 +261,10 @@ pub fn expand_sample_profile(payload: &[u8]) -> Result<Vec<Vec<u8>>, ProfileErro
             }
         });
 
+        if new_profile.profile.samples.is_empty() {
+            return Err(ProfileError::NotEnoughSamples);
+        }
```
Comment (Zylphrex) [inline relay-profiling/src/sample.rs]: This is going to drop the entire profile when one of the transactions in it has no samples right?

File: relay-profiling/src/outcomes.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+pub fn discard_reason(err: ProfileError) -> DiscardReason {
+    match err {
+        ProfileError::CannotSerializePayload => DiscardReason::FailedSerialization,
+        ProfileError::NotEnoughSamples => DiscardReason::NotEnoughSamples,
+        ProfileError::NoTransactionAssociated => DiscardReason::NoTransactionAssociated,
+        ProfileError::InvalidTransactionMetadata => DiscardReason::InvalidTransactionMetadata,
+        ProfileError::MissingProfileMetadata => DiscardReason::InvalidProfileMetadata,
+        ProfileError::MalformedStacks => DiscardReason::MalformedStacks,
+        ProfileError::MalformedSamples => DiscardReason::MalformedSamples,
```
Comment (jjbayer) [inline relay-profiling/src/outcomes.rs]: Given that this is a 1:1 mapping, why not just implement a direct mapping from `ProfileError -> &'static str`?

File: relay-profiling/src/error.rs [hunk 3]
```diff
@@ -2,8 +2,8 @@ use thiserror::Error;
 
 #[derive(Debug, Error)]
 pub enum ProfileError {
-    #[error("invalid json in profile")]
-    InvalidJson(#[source] serde_json::Error),
+    #[error(transparent)]
+    InvalidJSON(#[from] serde_json::Error),
```
Comment (jjbayer) [inline relay-profiling/src/error.rs]: We usually keep acronyms camel case, see 

https://github.com/getsentry/relay/blob/4fc80841346a088c40bb4cd69165e80b5e8862f7/relay-kafka/src/producer/mod.rs#L38

and 

https://rust-lang.github.io/api-guidelines/naming.html

**Review by jjbayer — APPROVED** [review]
Two comments, apart from those this looks good to me!

File: relay-profiling/src/sample.rs (hunk 1 above)
Comment (phacops) [reply relay-profiling/src/sample.rs]: Not anymore.

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**